compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7934,6 +7926,10 @@ private BLangLiteral createByteLiteral(Location pos, Byte value) {
     }
 
     private BLangExpression createTypeCastExpr(BLangExpression expr, BType targetType) {
+        if (expr.type.tag == targetType.tag) {
```

Comment:
Shall we send a separate PR checking if this is a same type. There could be same tag for say user defined types but actual types are different.